### PR TITLE
backend/feat: Transit Buses

### PR DESCRIPTION
**Pull request by khuzema786**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/spec/API/FrfsTicket.yaml [hunk 1]
```diff
@@ -39,10 +40,6 @@ types:
     quoteId: Id FRFSQuote
     _type: FRFSQuoteType
     vehicleType: VehicleCategory
-    serviceTierType: Maybe ServiceTierType
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/API/FrfsTicket.yaml:42]: Not consumed by FE, Safe to Remove

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml [hunk 2]
```diff
@@ -142,17 +142,11 @@ FRFSQuote:
...
     quantity: Int
     stationsJson: Text
     routeStationsJson: Maybe Text
     discountsJson: Maybe Text
     vehicleType: VehicleCategory
-    serviceTierProviderCode: Maybe Text
-    serviceTierType: Maybe ServiceTierType
-    serviceTierShortName: Maybe Text
-    serviceTierLongName: Maybe Text
-    serviceTierDescription: Maybe Text
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml:155]: Maybe Columns, only used in Non ONDC Flow, so safe to delete

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml [hunk 3]
```diff
@@ -180,9 +174,6 @@ FRFSQuote:
   types:
     FRFSQuoteType:
       enum: "SingleJourney, ReturnJourney, Pass, SpecialFareSingleJourney"
-    # For Bus it would be one route per leg, but for Metro it can be more than one route with transit stations for a single Journey leg.
-    FRFSRoutes:
-      enum: "Bus (Id Route), Metro [Id Route]"
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml:185]: Not a good enum as even Bus can have multiple routes during Transits, so removed and changed the approach.

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml [hunk 4]
```diff
@@ -225,14 +216,8 @@ FRFSTicketBooking:
...
     fromStationId: Id Station
     toStationId: Id Station
-    routeId: Maybe FRFSRoutes
     quantity: Int
     vehicleType: VehicleCategory
-    serviceTierProviderCode: Maybe Text
-    serviceTierType: Maybe ServiceTierType
-    serviceTierShortName: Maybe Text
-    serviceTierLongName: Maybe Text
-    serviceTierDescription: Maybe Text
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/FrfsTicket.yaml:235]: Maybe Column, only used in Non ONDC Flow, so safe to delete

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/FRFS/OnSearch.hs [hunk 5]
```diff
@@ -133,23 +136,6 @@ mkQuotes dOnSearch ValidatedDOnSearch {..} DQuote {..} = do
 
   startStation <- QStation.findByStationCode dStartStation.stationCode >>= fromMaybeM (InternalError $ "Station not found: " <> dStartStation.stationCode)
   endStation <- QStation.findByStationCode dEndStation.stationCode >>= fromMaybeM (InternalError $ "Station not found: " <> dEndStation.stationCode)
-  routeId <-
```
Comment (khuzema786) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/FRFS/OnSearch.hs:136]: No more required, we will dump full route details directly into JSON value in `routeStationsJson`

**Review by hkmangla — APPROVED** [review]

**Pull request merged**